File: leetcode3/이진희/1837. Sum of Digits in Base K.java
/*
1. 아이디어:
   - 10진수로 이뤄진 n을 k진수로 변환 후, 각 자리수의 합 구하기
   - while문으로 직접 자릿수 계산

2. 시간복잡도: O(logN)

3. 자료구조/알고리즘: 진법 변환 원리, 시뮬레이션
*/

class Solution {
    public int sumBase(int n, int k) {

        int ans = 0;
        
        while(n>0) {
            int p = n%k;

            ans += p;
            n/=k;
        }

        return ans;
    }
}

File: leetcode3/변지협/v3/1837. Sum of Digits in Base K.py

'''
1. 아이디어 :
k진수로 변환하고, 각 자리수를 더한다.

2. 시간복잡도 :
o(log n)

3. 자료구조/알고리즘 :
'''
class Solution:
    def sumBase(self, n: int, k: int) -> int:
        s = ''
        while True:
            if n == 0:
                break
            r = n % k
            n = n // k
            s += str(r)
        
        print(s)

        return sum([int(i) for i in s])
            